ui/component/viewers/videoViewer/view.jsx

```diff
@@ -150,6 +151,23 @@ function VideoViewer(props: Props) {
   const isFirstRender = React.useRef(true);
   const playerRef = React.useRef(null);
 
+  const [history, setHistory] = usePersistedState('watch-history', []);
+  React.useEffect(() => {
+    if (!history[0] || !history[0].indexOf(claim.permanent_url) !== -1) {
+      if (!history || !history.length) {
+        setHistory([claim.permanent_url]);
+      } else {
+        if (history.find((entry) => entry === claim.permanent_url)) {
+          history.splice(history.indexOf(claim.permanent_url), 1);
```

Comment:
Mutating states ... usually not recommended.  Just slice it I guess.
https://stackoverflow.com/a/59568563/977819